src/coreComponents/mesh/generators/WellGeneratorBase.cpp

```diff
@@ -521,62 +525,64 @@ void WellGeneratorBase::mergePerforations( array1d< array1d< localIndex > > cons
+void WellGeneratorBase::logInternalWell() const
 {
   if( MpiWrapper::commRank( MPI_COMM_GEOSX ) != 0 )
   {
     return;
   }
 
-  std::cout << std::endl;
-  std::cout << "++++++++++++++++++++++++++" << std::endl;
-  std::cout << "WellGeneratorBase = " << getName() << std::endl;
-  std::cout << "MPI rank = " << MpiWrapper::commRank( MPI_COMM_GEOSX ) << std::endl << std::endl;
-  std::cout << "Number of well elements = " << m_numElems << std::endl;
-
+  TableData tableWellData;
   for( globalIndex iwelem = 0; iwelem < m_numElems; ++iwelem )
   {
-    std::cout << "Well element #" << iwelem << std::endl;
-    std::cout << "Coordinates of the element center: " << m_elemCenterCoords[iwelem] << std::endl;
-    if( m_nextElemId[iwelem] < 0 )
-    {
-      std::cout << "No next well element" << std::endl;
-    }
-    else
-    {
-      std::cout << "Next well element # = " << m_nextElemId[iwelem] << std::endl;
-    }
-    if( m_prevElemId[iwelem][0] < 0 )
+    std::optional< globalIndex > nextElement;
+    std::optional< globalIndex > prevElement;
+
+    if( m_nextElemId[iwelem] >= 0 )
     {
-      std::cout << "No previous well element" << std::endl;
+      nextElement =  m_nextElemId[iwelem];
     }
-    else
+
+    if( m_prevElemId[iwelem][0] >= 0 )
     {
-      std::cout << "Previous well element #" << m_prevElemId[iwelem][0] << std::endl;
-    }
-    for( globalIndex inode = 0; inode < m_numNodesPerElem; ++inode )
-    {
-      if( inode == 0 )
-      {
-        std::cout << "First well node: #" << m_elemToNodesMap[iwelem][inode] << std::endl;
-      }
-      else
-      {
-        std::cout << "Second well node: #" << m_elemToNodesMap[iwelem][inode] << std::endl;
-      }
+      prevElement =  m_prevElemId[iwelem][0];
     }
+
+    tableWellData.addRow( iwelem,
+                          m_elemCenterCoords[iwelem][0],
+                          m_elemCenterCoords[iwelem][1],
+                          m_elemCenterCoords[iwelem][2],
+                          prevElement,
+                          nextElement );
   }
 
-  std::cout << std::endl << "Number of perforations = " << m_numPerforations << std::endl;
+  string wellTitle = "InternalWellGenerator " + getName();
+  TableLayout tableWellLayout = TableLayout( {
+      TableLayout::ColumnParam{"Element no.", TableLayout::Alignment::right},
+      TableLayout::ColumnParam{"CoordX", TableLayout::Alignment::center},
+      TableLayout::ColumnParam{"CoordY", TableLayout::Alignment::center},
+      TableLayout::ColumnParam{"CoordZ", TableLayout::Alignment::center},
+      TableLayout::ColumnParam{"Prev\nElement", TableLayout::Alignment::right},
+      TableLayout::ColumnParam{"Next\nElement", TableLayout::Alignment::right},
+    }, wellTitle );
+
+  TableTextFormatter tableFormatter( tableWellLayout );
+  GEOS_LOG_RANK_0( tableFormatter.ToString( tableWellData ));
+}
 
+void WellGeneratorBase::logPerforationTable() const
+{
+  TableData tablePerfoData;
```

Comment:
I think that if you now register those two `TableData tablePerfoData` and `TableData tableWellData`, you can now trigger the display as a table on `stdout` and into another format (easy to parse) in a file using the same data.
And that way you would have solved the parsing of the logs issue.

Reply:
We can use TableCSVFormatter (csv output) with TableTextFormatter ( log output). There is an example to the genreral description of this PR